### PR TITLE
Sorting an Array of 0s 1s and 2s

### LINKED ISSUE #1768: Sort an array of 0s,1s and 2s
# Aim

Want to add a solution of "Sort an array of 0s,1s and 2s"
Kindly assign me this problem. I would like to add my solution to this problem in C++ easily and efficiently using DNF sorting.
Link to the problem: https://www.geeksforgeeks.org/sort-an-array-of-0s-1s-and-2s/

# Details
Given an array A[] consisting 0s, 1s and 2s. The task is to write a function that sorts the given array. The functions should put all 0s first, then all 1s and all 2s in last.
Examples: 
 
Input: {0, 1, 2, 0, 1, 2}
Output: {0, 0, 1, 1, 2, 2}

Input: {0, 1, 1, 0, 1, 2, 1, 2, 0, 0, 0, 1}
Output: {0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 2, 2}

### Programming language

- [ ] C
- [ ] C++
- [ ] Java
- [ ] Python

### PR DESCRIPTION
**Pull request by Vidhi6340**
# Description

Please include a summary of the change and which issue is fixed. List any dependencies that are required for this change.

Fixes #1768 

issue number 1768 is fixed in this PR.

Code is added in algorithms/searching and sorting/C folder.
Test cases also have been added.

### CODE REVIEW

File: Algorithms/Searching and Sorting/C/sort_012_arr.c [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            arr[j]=arr[k];
+            arr[k]=temp;
+            k++;
+        }
+    }
+  }
+  printf("Sorted Array:");
+  for(i=0;i<N;i++)
+    printf("%d",arr[i]);
+
```
Comment (sameeksharl) [inline Algorithms/Searching and Sorting/C/sort_012_arr.c:30]: remove unused lines (don't leave empty lines)

File: Algorithms/Searching and Sorting/C/sort_012_arr.c [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+return 0;
+}
+
+
```
Comment (sameeksharl) [inline Algorithms/Searching and Sorting/C/sort_012_arr.c:35]: Do add space and time complexity

**Comment by sameeksharl** [comment]
@Vidhi6340 look into this issues and later it will be reviewed

**Comment by Vidhi6340** [comment]
> @Vidhi6340 look into these issues and later it will be reviewed

I have made the required changes. Please look into it.

**Comment by sameeksharl** [comment]
@Vidhi6340 merged into main

**Pull request merged**